src/Mod/Part/App/TopoShapeExpansion.cpp

```diff
@@ -2362,4 +2398,230 @@ bool TopoShape::fixSolidOrientation()
     return false;
 }
 
+TopoShape&
+TopoShape::makeElementBoolean(const char* maker, const TopoShape& shape, const char* op, double tolerance)
+{
+    return makeElementBoolean(maker, std::vector<TopoShape>(1, shape), op, tolerance);
+}
+
+
+TopoShape& TopoShape::makeElementBoolean(const char* maker,
```

Comment:
Future work: probably every one of these opcode should be its own method, this one is way too long.